### PR TITLE
👩‍💻 Add user registration

### PR DESCRIPTION
**Pull request by kenessajr**
Created `user registration route` and `tested`. fix linting on `test_country.py and test_profiles.py`. I also set `user active to False by default` on a user creation. We will activate the user once he confirms his registration.

### CODE REVIEW

**Comment by erwagasore** [comment]
Please also add the `photo` to the profiles model and make sure it is working

File: api/app/users/routes.py [hunk 1]
```diff
@@ -66,3 +66,19 @@ def list():
...
 def get_token():
     return APIResult({'token': g.user.generate_auth_token()})
+
+
+@users.route('/registration', methods=['POST'])
+@validate_with(create_user)
+def registration():
+    user = User()
+    user.import_data(request.method, request.get_json())
+    user.active = False
```
Comment (erwagasore) [inline api/app/users/routes.py]: Another question why do you want 'active=False'. What are you trying to achieve by doing that

File: api/app/users/test_users.py [hunk 2]
```diff
@@ -162,3 +162,12 @@ def test_users_routes(self, db_session, credentials, super_credentials):
...
         assert resp.status_code == 200
+
+        # post all required fields for the user
+        data = dict(username='foo', email='foo@bar.baz', password='foo', force=True)
+        resp = self.client.post(
+            url_for('users.registration'), data=json.dumps(data),
+            content_type='application/json')
+        assert resp.status_code == 201
+        assert 'self_url' in resp.json.keys()
+        assert 'profile_url' in resp.json.keys()
```
Comment (erwagasore) [inline api/app/users/test_users.py]: To better check for the creation of profile please query it from database.

File: api/app/users/routes.py (hunk 1 above)
Comment (kenessajr) [reply api/app/users/routes.py]: Remember that when we register a user he will have to first confirm his email to be active.

Comment (erwagasore) [reply api/app/users/routes.py]: Yes. But we will use `comfirmed_at` instead of `active`

Comment (kenessajr) [reply api/app/users/routes.py]:
```suggestion

```

**Pull request merged**